accelerator/common_core.c

```diff
@@ -505,27 +505,38 @@ status_t ta_get_bundle_by_addr(const iota_client_service_t* const service, flex_
   transaction_array_t* obj_res = transaction_array_new();
 
   if (txn_req == NULL || txn_res == NULL || obj_req == NULL || obj_res == NULL) {
+    log_error(cc_logger_id, "[%s:%d:%s]\n", __func__, __LINE__, "SC_TA_OOM");
     ret = SC_TA_OOM;
     goto done;
   }
 
   find_transactions_req_address_add(txn_req, addr);
 
   if (iota_client_find_transactions(service, txn_req, txn_res) != RC_OK) {
+    log_error(cc_logger_id, "[%s:%d:%s]\n", __func__, __LINE__, "SC_CCLIENT_FAILED_RESPONSE");
     ret = SC_CCLIENT_FAILED_RESPONSE;
     goto done;
   }
 
-  hash243_queue_push(&obj_req->hashes, find_transactions_res_hashes_get(txn_res, 0));
+  if (hash243_queue_count(txn_res->hashes) > 0) {
+    hash243_queue_push(&obj_req->hashes, find_transactions_res_hashes_get(txn_res, 0));
+  } else {
+    log_error(cc_logger_id, "[%s:%d:%s]\n", __func__, __LINE__, "SC_MAM_NOT_FOUND");
+    ret = SC_MAM_NOT_FOUND;
+    goto done;
+  }
+
   ret = ta_find_transaction_objects(service, obj_req, obj_res);
   if (ret != SC_OK) {
+    log_error(cc_logger_id, "[%s:%d:%X]\n", __func__, __LINE__, ret);
     goto done;
   }
 
   iota_transaction_t* curr_tx = transaction_array_at(obj_res, 0);
   flex_trits_to_trytes(bundle_hash, NUM_TRYTES_BUNDLE, transaction_bundle(curr_tx), NUM_TRITS_BUNDLE, NUM_TRITS_BUNDLE);
   ret = ta_get_bundle(service, bundle_hash, bundle);
   if (ret != SC_OK) {
+    log_error(cc_logger_id, "[%s:%d:%X]\n", __func__, __LINE__, ret);
```

Comment:
I have no idea why `ret` is dump via `%X`. Can you use something more verbose?

Reply:
I thought it would be more easier to find the error cases in `errors.h`, since we gave the `status_t` values in `errors.h` in hexademical. I can use `%d` instead of `%X` .

Reply:
You shall improve the message string for better understanding.